src/Microsoft.Azure.NotificationHubs/XiaomiRegistrationDescription.cs

```diff
@@ -0,0 +1,125 @@
+        public XiaomiRegistrationDescription(string xiaomiRegistrationId)
+            : this(string.Empty, xiaomiRegistrationId, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XiaomiRegistrationDescription"/> class.
+        /// </summary>
+        /// <param name="xiaomiRegistrationId">The Xiaomi registration identifier.</param>
+        /// <param name="tags">The tags.</param>
+        public XiaomiRegistrationDescription(string xiaomiRegistrationId, IEnumerable<string> tags)
+            : this(string.Empty, xiaomiRegistrationId, tags)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XiaomiRegistrationDescription"/> class.
+        /// </summary>
+        /// <param name="sourceRegistration">The source registration.</param>
+        public XiaomiRegistrationDescription(XiaomiRegistrationDescription sourceRegistration)
+            : base(sourceRegistration)
+        {
+            XiaomiRegistrationId = sourceRegistration.XiaomiRegistrationId;
+        }
+
+        internal XiaomiRegistrationDescription(string notificationHubPath, string xiaomiRegistrationId, IEnumerable<string> tags)
+            : base(notificationHubPath)
+        {
+            if (xiaomiRegistrationId == null)
+            {
+                throw new ArgumentNullException("xiaomiRegistrationId");
+            }
+
+            XiaomiRegistrationId = xiaomiRegistrationId;
+
+            if (tags != null)
+            {
+                Tags = new HashSet<string>(tags);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the Xiaomi registration identifier.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The Xiaomi registration identifier.
+        /// </returns>
+        [DataMember(Name = ManagementStrings.XiaomiRegistrationId, Order = 2001, IsRequired = true)]
+        public string XiaomiRegistrationId { get; set; }
+
+        internal override string AppPlatForm
+        {
+            get
+            {
+                // This needs to be changed to XiaomiCredential.AppPlatformName in a following PR when we add XiaomiCredential class
+                return "xiaomi"; 
+            }
+        }
+
+        internal override string RegistrationType
+        {
+            get
+            {
+                // This needs to be changed to XiaomiCredential.AppPlatformName in a following PR when we add XiaomiCredential class
+                return "xiaomi"; 
+            }
+        }
+
+        internal override string PlatformType
+        {
+            get
+            {
+                // This needs to be changed to XiaomiCredential.AppPlatformName in a following PR when we add XiaomiCredential class
+                return "xiaomi"; 
+            }
+        }
+
+        internal override void OnValidate()
+        {
+            if (string.IsNullOrWhiteSpace(XiaomiRegistrationId))
```

Comment:
Ah actually this is a good question. I only have one xiaomi deviceToken but if we see a pattern or something that we should pay attention to, we should definitely put it into the onValidate method

Reply:
Closing this one since I will build a special nuget package and publish it with a special version number with my own repository locally and let the partner team use it.